R/namespace.R

```diff
@@ -10,16 +10,29 @@ namespace_imports <- function(path = find_package()) {
 
   full_imports <- lengths(imports) == 1L
 
-  # this loads the namespaces, but is the easiest way to do it
-  imports[full_imports] <- lapply(imports[full_imports], function(x) list(x, getNamespaceExports(asNamespace(x))))
+  imports[full_imports] <- lapply(imports[full_imports], safe_get_exports)
 
+  funs <- lapply(imports, `[[`, 2L)
   data.frame(
-    pkg = unlist(lapply(imports, function(x) rep(x[[1L]], length(x[[2L]])))),
-    fun = unlist(lapply(imports, `[[`, 2L)),
+    pkg = rep(unlist(lapply(imports, `[[`, 1L)), lengths(funs)),
```

Comment:
Use `rep(full_imports, lengths(funs))` here after refactoring `safe_get_exports()` to return only the functions?

Reply:
Sorry, not seeing it...

(1) we get a list with elements like (a) `list(ns, imported_funs)` or (b) `list(ns)`
(2) `safe_get_exports()` fills out the (b)-type elements to match the style of (a)-type
(3) combine to get the output data.frame

the presence of the (a)-type elements makes it harder to refactor, but I may be missing something.

Reply:
ah nevermind, I didn't see the named import part because it wasn't in the diff.

Reply:
Reworked now, PTAL.